Not very autonomous, just people working hard

realKfiros opened the pull request:
Add a lot of stuff
~Dare we say finish the app?~

realKfiros committed:
pages: Add autonomous page

guy-av commented on lib/pages/autonomous_page.dart:
```diff
@@ -0,0 +1,148 @@
+import 'package:neatteam_scouting_2023/models/cycle.dart';
+import 'package:neatteam_scouting_2023/styles/style_form_field.dart';
+
+const changeStateColors = <ChargeStationState, Color>{
+  ChargeStationState.failed: Colors.redAccent,
+  ChargeStationState.engaged: Colors.blueAccent,
+  ChargeStationState.docked: Colors.greenAccent
+};
```
Move to the state

guy-av commented on lib/pages/autonomous_page.dart:
```diff
@@ -0,0 +1,148 @@
+    });
+  }
+
+  Widget buildSegment(String text) {
```
```suggestion
  Widget _buildSegment(String text) {
```

guy-av commented on lib/pages/autonomous_page.dart:
```diff
@@ -0,0 +1,148 @@
+    model.chargeStationState = ChargeStationState.failed;
+  }
+
+  void setChargeStationState(ChargeStationState css) {
```
```suggestion
  void _setChargeStationState(ChargeStationState css) {
```

realKfiros committed:
widgets: Add cycles list

realKfiros committed:
widgets: Add in game action bar and fouls

realKfiros committed:
pages: Update home page for the new match provider

realKfiros committed:
pages: Update match page for the new match provider

realKfiros committed:
lib: Whole lotta changes

guy-av committed:
utils: Add match param to MatchState

guy-av committed:
widgets: Add sliding segmented control

guy-av committed:
pages: cycles: Overhaul

- Replace dropdowns with segment controls
- Remove unnecessary fields
- Add auto half cycle to teleop and auto pages
- Add fail button to cycle page
- Pop cycle page when grid level and zone are filled
- Add timer to cycle page

guy-av committed:
pub: Add stop_watch_timer

guy-av committed:
pages: Automatically add new cycle when one is finished

guy-av committed:
models: Default game piece in robot (auto) to false

guy-av committed:
models: Add match finished indicator

guy-av committed:
widgets: action_bar: Add finish match button

guy-av committed:
pages: team_info: Navigate straight to auto page

guy-av committed:
pages: auto: Add navigation to teleop page

- Disable segmented control if robot did not attempt chatge station

guy-av committed:
pages: cycles: Rename Autonomous context title to Auto

guy-av committed:
pages: teleop: Add navigation to endgame

- Make first cycle half if robot has game piece on auto finish

guy-av committed:
pages: endgame: Add stopwatch

guy-av committed:
pages: team_info: Add scouter name field

guy-av committed:
csv: Fix cycles data generation

- Add test for cycles csv

guy-av committed:
pub: Add uuid and archive

guy-av committed:
utils: Share both CSVs in zip archive

guy-av committed:
pages: cycles: Remove some options when auto

guy-av committed:
pub: Add shared preferences

guy-av committed:
pub: Add json_serializable, json_annotation and build_runner

guy-av committed:
models: Add json serialization

guy-av committed:
Add storage to save state to disk

guy-av committed:
pages: cycles: Hide pickup zone when half cycle

guy-av committed:
pub: Add flutter_slidable

guy-av committed:
pages: home: Allow removal of matches

guy-av committed:
lib: Fix analyze issues (Mostly remove unused imports)

guy-av committed:
pages: team_info: Use segmented control instead of dropdowns

guy-av committed:
ios: Setup

guy-av committed:
pub: Decrease version to 0.1.0+1

guy-av committed:
models: match: Add did die field

guy-av committed:
pages: home: Add delete all matches button

guy-av committed:
pages: cycle: Enable changing is half

guy-av committed:
pages: Add notes

guy-av committed:
pages: cycle: Fix finished cycle going back on change

guy-av committed:
models: g: Generate didDie and notes for matches

guy-av committed:
csv: Add match notes and did die

guy-av committed:
ios: Setup signing

guy-av committed:
--wip--
